trunk_lexer/src/lexer.rs

```diff
@@ -158,356 +122,161 @@ impl Lexer {
     }
 
     fn scripting(&mut self) -> Result<Token, LexerError> {
-        // We should never reach this point since we have the empty checks surrounding
-        // the call to this function, but it's better to be safe than sorry.
-        if self.peek.is_none() {
-            return Err(LexerError::UnexpectedEndOfFile);
-        }
-
-        // Since we have the check above, we can safely unwrap the result of `.next()`
-        // to help reduce the amount of indentation.
-        self.next();
-        let char = self.current.unwrap();
-
-        let kind = match char {
-            b'@' => {
+        let kind = match self.peek_buf() {
```

Comment:
I kinda wonder what this type of `match` actually compiles down to, but assume we're not too worried about performance at this stage. It may be worth revisiting later.

It makes the code pretty nice IMHO.

Reply:
If you were truly interested, you could look at similar code in https://godbolt.org/ for the assembly output.

Reply:
Yep, I do intend to at some stage. But for now I think I'm happy with the nicer code.